Added reinforcement learning task description

02shanks opened the pull request:
Partially fixes huggingface/huggingface.js#326. Added task description.

merveenoyan commented on tasks/src/Reinforcement-learning/about.md:
```diff
@@ -0,0 +1,9 @@
+
+### Robotics
+
+Reinforcement learning can be used to teach robots how to perform tasks by providing them with feedback about their performance. For example, a robot might be given a score for each task it completes, and the robot can use reinforcement learning to learn how to improve its performance by trial and error.
```
Can you explain a bit more about this and avoid using passive voice? 🙂

merveenoyan commented on tasks/src/Reinforcement-learning/about.md:
```diff
@@ -0,0 +1,9 @@
+
+### Gaming
+
+Reinforcement learning is used in gaming to help agents learn how to optimally play a game. This is done by providing the agent with feedback in the form of rewards or punishments after each action is taken. Over time, the agent will learn which actions lead to the highest rewards and will adjust its behaviour accordingly.
```
Can you elaborate a little more and avoid using passive voice and maybe explain what an agent is? 🙂

merveenoyan reviewed (commented):
Thanks a lot for working on this! 🤗 I provided very general feedback, please take a look. 👀

02shanks commented:
@merveenoyan  Can you please review the changes!!

merveenoyan commented on tasks/src/Reinforcement-learning/about.md:
```diff
@@ -0,0 +1,9 @@
+
+### Gaming
+
+Reinforcement learning is used in gaming to provide game characters with a set of possible actions and then reward them for selecting the optimal action. This allows game characters to learn from their mistakes and eventually become better at choosing the optimal actions.
```
```suggestion
Reinforcement learning is used in video games to provide non-playable characters with a set of possible actions. The trained model then rewards them for selecting the optimal action. This allows non-playable characters to learn from their mistakes and eventually become better at choosing the optimal actions.
```

simoninithomas replied:
In game industry we don't use RL as replace of NPC (non playable characters) we use FSM (finite state machine), Behavior Tree or GOAP settings.

But I'm agree that's a good explanation so we can keep this way.

merveenoyan commented on tasks/src/Reinforcement-learning/about.md:
```diff
@@ -0,0 +1,9 @@
+
+### Robotics
+
+Reinforcement learning is used in robotics to enable robots to learn from experience and improve their performance over time. For example, a robot might be programmed to clean a floor by starting in one corner and moving in a straight line until it reaches the other corner. If the robot encounters an obstacle, it will learn from this experience and try a different route the next time it cleans th ...
```
```suggestion
Reinforcement learning is used in robotics to enable robots to learn from experience and improve their performance over time. For example, a robot might be programmed to clean a floor by starting in one corner and moving in a straight line until it reaches the other corner. If the robot encounters an obstacle, it will learn from this experience and try a different route the next time it cleans the floor.
```
Does this really apply, the robots have sensors no? I don't think RL is needed for this. I'd try not to give too much detail about it. WDYT @simoninithomas

simoninithomas replied:
It depends on the setting but I would change to:

Reinforcement learning is used in robotics to enable robots to learn from experience and improve their performance over time. For example, a robot might **have the goal to clean** a floor by starting in one corner and moving in a straight line until it reaches the other corner. If the robot encounters an obstacle, it will learn from this experience and try a different route the next time it cleans the floor.

merveenoyan reviewed (approved):
Overall LGTM, I'd like @simoninithomas to take a look at the robotics use case though and then we can merge 🙂

merveenoyan commented:
@02shanks I'd keep only the one that's about gaming one rather than robotics one since another contributor has contributed self-driving cars use cases in depth which is similar to this one 🙂 and then we can merge. 
Can you remove `data.ts` and bring back empty sections in placeholder file?

02shanks commented:
@merveenoyan Can you please review the changes!!

02shanks commented:
@merveenoyan Can you please review the changes!!

02shanks commented:
@merveenoyan I tried to make changes to resolve the merge conflict, are they right?

merveenoyan commented:
@02shanks ![image](https://user-images.githubusercontent.com/53175384/197626795-86e76e77-adfb-4ca0-af0e-7c0a4a3ab2da.jpeg)
In data.ts file you've left both versions of the file so the merge conflict is still there. 😞 
Can you copy your changes somewhere and write your changes on top of the version in main branch? From what I see what you've did was copy and paste all of your changes and replacing it with what is pulled, which makes it very hard to review (hard to see what has changed). Normally I should be seeing the changes in the lines that are added, but now I see the whole page being removed and your changes being appended. (You can see below how it looks to me)
![image](https://user-images.githubusercontent.com/53175384/197627377-12286cf2-026c-4f3d-af3e-c0d5b3feb7a2.jpeg)

02shanks commented:
@merveenoyan sorry I messed up on my last step to resolve the conflict. I kept both changes and incorporated them into the final merge (I should have removed my changes and added them in a later commit). Sorry for the inconvenience from my side. What should I do next?

02shanks commented:
@merveenoyan do I have to make any changes from my side?

merveenoyan commented:
@02shanks very sorry for late reply, can you fix merge conflicts and I will merge afterwards 🙂

02shanks commented:
Hello @merveenoyan,
I hope you're doing well. I wanted to follow up on the PR that I submitted. I have resolved the conflict and I think it's ready for review.I am extremely sorry for late contribution to solve conflict. Could you please take a look and let me know if there is anything else needs to be done before we can merge it? I can see there has been many changes. If you can suggest me what can I add then I can try to add asap.
Thank you for your time and efforts.

merveenoyan commented:
@02shanks no worries, it's all about contributions! I can't see what's different compared to main though, is it possible that you accidentally removed your changes while solving merge conflict?

02shanks commented:
@merveenoyan I did remove my changes to resolve conflict. Now I have added my changes back. Can you review it and suggest changes?

merveenoyan commented on tasks/src/reinforcement-learning/about.md:
```diff
@@ -1,72 +1,72 @@
+
+### Gaming
+
+Reinforcement learning is a type of machine learning that allows a computer or artificial intelligence (AI) to learn how to play a game by receiving rewards or punishments for certain actions. The AI uses this feedback to improve its strategy over time. In a game, the AI learns to make decisions by choosing actions that will lead to the most rewards, such as winning the game or achieving a high s ...
```
I think it would be nice if you could not remove what previous contributor contributed but rather on top of it 🙂

02shanks replied:
@merveenoyan I made changes!!

merveenoyan reviewed (commented):
Left a comment 🙂

02shanks commented:
@merveenoyan I guess I mess up while reverting back to my previous commit and add my changes on top on present definition.
Should I close this PR and create new PR and add my changes or can you suggest how can clean this mess?

02shanks commented:
![Screenshot from 2023-01-20 02-37-15](https://user-images.githubusercontent.com/76114538/213559675-b1fc5196-98cf-46a4-bf35-28d3752bcd32.png)
I am constantly receiving this. How can I fix this?

merveenoyan commented:
@02shanks sorry for delay, I think it should work better if you close this one and add changes if solving merge conflicts is not convenient for you 😊
